explore_simulation_approach/3a_check_simulated_experiments.py

```diff
@@ -0,0 +1,332 @@
+
+# Pickle files saving specific and generic gene ids
+template_specific_gene_ids_filename = params["template_specific_gene_ids_filename"]
+generic_gene_ids_filename = "generic_gene_ids.pickle"
+
+# Identifier for simulated experiment
+i = 7
+
+# File to scaler transform data to 0-1 range
+scaler_transform_filename = params["scaler_filename"]
+# -
+
+# ## Check clustering
+
+simulated_filename = os.path.join(
+    local_dir,
+    "pseudo_experiment",
+    f"selected_simulated_data_{project_id}_{i}.txt",
+)
+
+simulated_experiment = pd.read_csv(simulated_filename, sep="\t", index_col=0, header=0)
+
+print(simulated_experiment.shape)
+simulated_experiment.head()
+
+# +
+# Get common genes
+# Get specific genes
+
+# Load pickled file
+with open(template_specific_gene_ids_filename, "rb") as specific_fh:
+    specific_gene_ids = pickle.load(specific_fh)
+
+with open(generic_gene_ids_filename, "rb") as generic_fh:
+    generic_gene_ids = pickle.load(generic_fh)
+# -
+
+# Get NA genes
+all_gene_ids = simulated_experiment.columns
+all_gene_ids_tmp = all_gene_ids.difference(specific_gene_ids)
+na_gene_ids = all_gene_ids_tmp.difference(generic_gene_ids)
+
+# Simulated data subsets
+simulated_specific_df = simulated_experiment[specific_gene_ids]
+simulated_common_df = simulated_experiment[generic_gene_ids]
+simulated_na_df = simulated_experiment[na_gene_ids]
+
+print(simulated_specific_df.shape)
+simulated_specific_df
+
+print(simulated_common_df.shape)
+simulated_common_df
+
+print(simulated_na_df.shape)
+simulated_na_df
+
+f = sns.clustermap(simulated_specific_df.T, cmap="viridis")
+f.fig.suptitle("Simulated experiment specific genes")
+
+f = sns.clustermap(simulated_common_df.T, cmap="viridis")
+f.fig.suptitle("Simulated experiment common genes")
+
+f = sns.clustermap(simulated_na_df.T, cmap="viridis")
+f.fig.suptitle("Simulated experiment NA genes")
+
+# ## Check latent space distribution
+
+normalized_compendium = pd.read_csv(
+    normalized_compendium_filename, sep="\t", index_col=0, header=0
+)
+
+# +
+# Files
+model_encoder_filename = glob.glob(os.path.join(vae_model_dir, "*_encoder_model.h5"))[0]
+
+weights_encoder_filename = glob.glob(
+    os.path.join(vae_model_dir, "*_encoder_weights.h5")
+)[0]
+
+model_decoder_filename = glob.glob(os.path.join(vae_model_dir, "*_decoder_model.h5"))[0]
+
+weights_decoder_filename = glob.glob(
+    os.path.join(vae_model_dir, "*_decoder_weights.h5")
+)[0]
+
+# Load saved models
+loaded_model = load_model(model_encoder_filename)
+loaded_decode_model = load_model(model_decoder_filename)
+
+loaded_model.load_weights(weights_encoder_filename)
+loaded_decode_model.load_weights(weights_decoder_filename)
+# -
+
+# Encode normalized expression data
+compendium_encoded = loaded_model.predict_on_batch(normalized_compendium)
+compendium_encoded_df = pd.DataFrame(
+    compendium_encoded, index=normalized_compendium.index
+)
+
+# +
+# PCA embedding of VAE encoded data
+
+random_state = 1
```

Comment:
Looks like you didn't use this?

Reply:
This was from when i was previously using umap to visualize the data, I don't need this anymore